### PR TITLE
Add support to the client uses abstract address to connect to the daemon

### PR DESCRIPTION
**Pull request by daxinc**
*Description of changes:*

The current implementation assumes the client connects to the daemon using file system based socket. If the client uses [an abstract socket address], it fails to send back response.

The change allows the daemon to communicate back to the client irrespective to the type of socket it uses.

### CODE REVIEW

File: clock-bound-d/Cargo.toml [hunk 1]
```diff
@@ -20,11 +20,12 @@ clap = "2.33"
...
+uds = "0.2.6"
 
 [[bin]]
 name = "clockboundd"
 path = "src/main.rs"
 doc = false
 
 [badges]
-github = { repository = "aws/clock-bound-d"}
+github = { repository = "aws/clock-bound"}
```
Comment (basilnsindhu) [inline clock-bound-d/Cargo.toml:31]: Why are we changing this to `aws/clock-bound`? We have the badge set to `-c` and `-d` based on the client and daemon

Comment (dfoxfranke) [reply clock-bound-d/Cargo.toml:31]: I do recall that there was some issue here; is this meant to fix it? In any case please rebase unrelated changes like this into a separate commit.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+target/
+.idea/
+Cargo.lock
```
Comment (dfoxfranke) [inline .gitignore:3]: Why are you adding `Cargo.lock` to .gitignore? This should remain checked-in for program crates.

File: clock-bound-d/Cargo.toml (hunk 1 above)
Comment (daxinc) [reply clock-bound-d/Cargo.toml:31]: The issue is if you put it in the browser, it is 404. Ideally, those are 2 repos.

File: .gitignore (hunk 2 above)
Comment (daxinc) [reply .gitignore:3]: It is in the `.gitignore` of the `clock-bound-c` and `clock-bound-d` folder before they are merged into one.

Comment (daxinc) [reply .gitignore:3]: Will remove it.

File: clock-bound-d/Cargo.toml (hunk 1 above)
Comment (jacoblwisniewski) [reply clock-bound-d/Cargo.toml:31]: Haven't tested this, but I'm pretty sure we can remove this entirely. When writing this I attempted to add badges like this, but instead tied them into the readme generation. They are now generated via https://github.com/aws/clock-bound/blob/main/clock-bound-d/README.tpl#L1 and this line I believe doesn't do anything anymore. See clock-bound-c, this line doesn't exist: https://github.com/aws/clock-bound/blob/main/clock-bound-c/Cargo.toml

The repository info on crates.io gets pulled from https://github.com/aws/clock-bound/blob/main/clock-bound-d/Cargo.toml#L9.

**Comment by daxinc** [comment]
Update the PR so that it only includes the changes related to the feature.

File: clock-bound-d/src/server.rs [hunk 3]
```diff
@@ -56,7 +57,7 @@ impl ClockBoundServer {
     ) -> Result<(), io::Error> {
         let mut request: [u8; 12] = [0; 12];
 
-        let (request_size, client) = match self.socket.recv_from(&mut request) {
+        let (request_size, client) = match self.socket.recv_from_unix_addr(&mut request) {
```
Comment (dfoxfranke) [inline clock-bound-d/src/server.rs:60]: You can use `?` notation to do this more idomatically, as simply `let (request_size_client) = self.socket.recv_from_unix_addr(&mut request)?;`

File: clock-bound-d/src/server.rs [hunk 4]
```diff
@@ -78,15 +79,9 @@ impl ClockBoundServer {
...
-        if let Some(client_path) = client.as_pathname() {
-            match self.socket.send_to(&mut response, client_path) {
-                Ok(..) => return Ok(()),
-                Err(e) => warn!("Failed to send response to client. Error: {:?}", e),
-            }
-        } else {
-            // Getting the pathname may fail if an unbound socket tries to communicate with the
-            // daemon. In that case, we will drop the packet instead of attempting to reply.
-            warn!("Failed to get client pathname. Dropping packet.");
+        match self.socket.send_to_unix_addr(&mut response, &client) {
```
Comment (dfoxfranke) [inline clock-bound-d/src/server.rs:82]: It's a little messy to be returning from one branch here and not the other, when you're just returning on the next line anyway. How about just `if let  Err(e) = ... { warn!(...) }`?

**Comment by dfoxfranke** [comment]
This now looks correct. I noted a couple stylistic nits. Please address those and then rebase into a single commit.

**Comment by daxinc** [comment]
> This now looks correct. I noted a couple stylistic nits. Please address those and then rebase into a single commit.

@dfoxfranke

Made the stylistic changes you pointed out. Can you merged the PR using `Squash and Merged` so that I don't have to make it a single commit?

**Pull request merged**